akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java

```diff
@@ -489,6 +490,53 @@ void groupByExample() {
     // #groupBy
   }
 
+  void watchTerminationExample() {
+    // #watchTermination
+    List<IntSupplier> elements = Arrays.asList(() -> 1, () -> 2, () -> 3);
+    Source.from(elements)
+        // discard the materialized value of the stream as we're not interested in it
+        .watchTermination(Keep.none())
+        // we can also use Keep.right, Keep.left, or Keep.both depending on our needs wrt the
+        // materialized value
+        .runForeach(element -> System.out.println(element.getAsInt()), system)
+        .whenComplete(
+            (done, exc) -> {
+              if (done != null) System.out.println("Done");
+              else System.out.println(exc.getMessage());
+            });
```

Comment:
This is not how you would use `watchTermination`, it creates a termination watching operators but immediately discards the `CompletionStage` that it signals termination through.

What this example currently does is adding a reaction to the materialized value of type `CompletionStage` that `Sink.foreach` creates.

How it should be used is to either set up a callback inline in the lambda passed to `watchTermination((previousMatVal, streamTermination) => ...)` or you'd make sure to keep it through out the chain of operators and extract it in the end. I think the former is cleanest to showcase in a sample.

It's not very useful with `watchTermination` in a finite simple flow like this but I think it is nonetheless good to keep it this simple.

Reply:
You're totally right, my bad. I fixed the samples moving the prints to the lambda of `watchTermination`.